### ISSUE TITLE
[Crontab Error ]File "/usr/lib/python2.7/getpass.py", line 135, in _raw_input     raise EOFError

### ISSUE THREAD

**Issue by apanly** [fabric/fabric#1230]
I have a Job deployed on Ubuntu's Crontab,But Some Servers allways run into an error.
# 

File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 647, in host_prompting_wrapper
    return func(_args, *_kwargs)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/operations.py", line 1042, in run
    shell_escape=shell_escape)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/operations.py", line 909, in _run_command
    channel=default_channel(), command=wrapped_command, pty=pty,
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/state.py", line 391, in default_channel
    chan = _open_session()
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/state.py", line 382, in _open_session
    return connections[env.host_string].get_transport().open_session()
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 159, in __getitem__
    self.connect(key)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 151, in connect
    user, host, port, cache=self, seek_gateway=seek_gateway)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 531, in connect
    password = prompt_for_password(text)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 612, in prompt_for_password
    new_password = _password_prompt(password_prompt, stream)
  File "/home/www/virpython/local/lib/python2.7/site-packages/fabric/network.py", line 584, in _password_prompt
    return getpass.getpass(prompt.encode('ascii', 'ignore'), stream)
  File "/usr/lib/python2.7/getpass.py", line 83, in unix_getpass
    passwd = fallback_getpass(prompt, stream)
  File "/usr/lib/python2.7/getpass.py", line 118, in fallback_getpass
    return _raw_input(prompt, stream)
  File "/usr/lib/python2.7/getpass.py", line 135, in _raw_input
    raise EOFError
EOFError

**Comment by dalekurt** [comment]
What causes this error and how did you fix it. I have a Jenkins job running a python script and I get a similar error.

**Comment by edbrannin** [comment]
@dalekurt For other people who find this from Google:

If your Jenkins job is bombing with this error, double-check that it's providing the right password without interaction.

**Comment by bitprophet** [comment]
Yup, this is usually a result of the Python `getpass` library trying to run when not attached to a live terminal. As @edbrannin says you need to find a noninteractive method for dealing with this, e.g. in Fabric itself you can use the 'prompts' feature (see docs) or the 'password' option if this is just the login or key password.

**Comment by chill666** [comment]
I encountered same issue when trying to run my program through crontab on Mac OS. My program works fine when run it independently. Normally, I never set env.password for remote system password. For instead, I set env.key_filename. It works fine if i don't have to run sudo command. So ideally, it shouldn't prompt any password typing. 

By referring to above, i also tried to set environment variables to pass the password. Yet, i still get the same error. What did i miss? Anyone can help pls?

Thx.
